\doxysection{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test Class Reference}
\hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test}{}\label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}


Inheritance diagram for gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test\+:\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=233pt]{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test__inherit__graph}
\end{center}
\end{figure}


Collaboration diagram for gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test\+:\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=233pt]{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test__coll__graph}
\end{center}
\end{figure}
\doxysubsubsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_adcaa33ea772d54a56ac98131b0335bb4}{test\+Catches\+Cxx\+Exceptions\+In\+Fixture\+Constructor}} (self)
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a585b7072f518a99a8e5b0b16d7fa8ac6}{test\+Catches\+Cxx\+Exceptions\+In\+Fixture\+Destructor}} (self)
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_af67ed195d634b21459ca410bf3607524}{test\+Catches\+Cxx\+Exceptions\+In\+Set\+Up\+Test\+Case}} (self)
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a0c93888c01978d87a6c57c49b95f7624}{test\+Catches\+Cxx\+Exceptions\+In\+Tear\+Down\+Test\+Case}} (self)
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_ab23c20c23acb95d267f4a5ad7311ab5e}{test\+Catches\+Cxx\+Exceptions\+In\+Set\+Up}} (self)
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a6e65997bc7dd3700c29fee2e537a9d49}{test\+Catches\+Cxx\+Exceptions\+In\+Tear\+Down}} (self)
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_aae7ffa4b9583e66b08b41c5e7317d4a3}{test\+Catches\+Cxx\+Exceptions\+In\+Test\+Body}} (self)
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a69dd04c0f3eaf00738e0c3bfe8f31e68}{test\+Catches\+Non\+Std\+Cxx\+Exceptions}} (self)
\item 
\mbox{\hyperlink{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a5952b0a1b4fd09ba078635bf4e4bb1b1}{test\+Unhandled\+Cxx\+Exceptions\+Abort\+The\+Program}} (self)
\end{DoxyCompactItemize}


\doxysubsection{Detailed Description}
\begin{DoxyVerb}Tests C++ exception-catching behavior.

   Tests in this test case verify that:
   * C++ exceptions are caught and logged as C++ (not SEH) exceptions
   * Exception thrown affect the remainder of the test work flow in the
     expected manner.
\end{DoxyVerb}
 

Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00112}{112}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.



\doxysubsection{Member Function Documentation}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_adcaa33ea772d54a56ac98131b0335bb4}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testCatchesCxxExceptionsInFixtureConstructor@{testCatchesCxxExceptionsInFixtureConstructor}}
\index{testCatchesCxxExceptionsInFixtureConstructor@{testCatchesCxxExceptionsInFixtureConstructor}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testCatchesCxxExceptionsInFixtureConstructor()}{testCatchesCxxExceptionsInFixtureConstructor()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_adcaa33ea772d54a56ac98131b0335bb4} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Catches\+Cxx\+Exceptions\+In\+Fixture\+Constructor (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00121}{121}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00121\ \ \ \textcolor{keyword}{def\ }testCatchesCxxExceptionsInFixtureConstructor(self):}
\DoxyCodeLine{00122\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'C++\ exception\ with\ description\ '}}
\DoxyCodeLine{00123\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'"{}Standard\ C++\ exception"{}\ thrown\ '}}
\DoxyCodeLine{00124\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'in\ the\ test\ fixture\(\backslash\)'s\ constructor'}}
\DoxyCodeLine{00125\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00126\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'unexpected'}\ \textcolor{keywordflow}{not}\ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT,}
\DoxyCodeLine{00127\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'This\ failure\ belongs\ in\ this\ test\ only\ if\ '}}
\DoxyCodeLine{00128\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'"{}CxxExceptionInConstructorTest"{}\ (no\ quotes)\ '}}
\DoxyCodeLine{00129\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'appears\ on\ the\ same\ line\ as\ words\ "{}called\ unexpectedly"{}'})}
\DoxyCodeLine{00130\ }

\end{DoxyCode}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a585b7072f518a99a8e5b0b16d7fa8ac6}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testCatchesCxxExceptionsInFixtureDestructor@{testCatchesCxxExceptionsInFixtureDestructor}}
\index{testCatchesCxxExceptionsInFixtureDestructor@{testCatchesCxxExceptionsInFixtureDestructor}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testCatchesCxxExceptionsInFixtureDestructor()}{testCatchesCxxExceptionsInFixtureDestructor()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a585b7072f518a99a8e5b0b16d7fa8ac6} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Catches\+Cxx\+Exceptions\+In\+Fixture\+Destructor (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00134}{134}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00134\ \ \ \ \ \textcolor{keyword}{def\ }testCatchesCxxExceptionsInFixtureDestructor(self):}
\DoxyCodeLine{00135\ \ \ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'C++\ exception\ with\ description\ '}}
\DoxyCodeLine{00136\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'"{}Standard\ C++\ exception"{}\ thrown\ '}}
\DoxyCodeLine{00137\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'in\ the\ test\ fixture\(\backslash\)'s\ destructor'}}
\DoxyCodeLine{00138\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00139\ \ \ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInDestructorTest::TearDownTestCase()\ '}}
\DoxyCodeLine{00140\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00141\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00142\ }

\end{DoxyCode}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_ab23c20c23acb95d267f4a5ad7311ab5e}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testCatchesCxxExceptionsInSetUp@{testCatchesCxxExceptionsInSetUp}}
\index{testCatchesCxxExceptionsInSetUp@{testCatchesCxxExceptionsInSetUp}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testCatchesCxxExceptionsInSetUp()}{testCatchesCxxExceptionsInSetUp()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_ab23c20c23acb95d267f4a5ad7311ab5e} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Catches\+Cxx\+Exceptions\+In\+Set\+Up (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00171}{171}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00171\ \ \ \textcolor{keyword}{def\ }testCatchesCxxExceptionsInSetUp(self):}
\DoxyCodeLine{00172\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'C++\ exception\ with\ description\ "{}Standard\ C++\ exception"{}'}}
\DoxyCodeLine{00173\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'\ thrown\ in\ SetUp()'}}
\DoxyCodeLine{00174\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00175\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInSetUpTest::TearDownTestCase()\ '}}
\DoxyCodeLine{00176\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00177\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00178\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInSetUpTest\ destructor\ '}}
\DoxyCodeLine{00179\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00180\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00181\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInSetUpTest::TearDown()\ '}}
\DoxyCodeLine{00182\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00183\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00184\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'unexpected'}\ \textcolor{keywordflow}{not}\ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT,}
\DoxyCodeLine{00185\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'This\ failure\ belongs\ in\ this\ test\ only\ if\ '}}
\DoxyCodeLine{00186\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'"{}CxxExceptionInSetUpTest"{}\ (no\ quotes)\ '}}
\DoxyCodeLine{00187\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'appears\ on\ the\ same\ line\ as\ words\ "{}called\ unexpectedly"{}'})}
\DoxyCodeLine{00188\ }

\end{DoxyCode}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_af67ed195d634b21459ca410bf3607524}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testCatchesCxxExceptionsInSetUpTestCase@{testCatchesCxxExceptionsInSetUpTestCase}}
\index{testCatchesCxxExceptionsInSetUpTestCase@{testCatchesCxxExceptionsInSetUpTestCase}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testCatchesCxxExceptionsInSetUpTestCase()}{testCatchesCxxExceptionsInSetUpTestCase()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_af67ed195d634b21459ca410bf3607524} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Catches\+Cxx\+Exceptions\+In\+Set\+Up\+Test\+Case (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00143}{143}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00143\ \ \ \textcolor{keyword}{def\ }testCatchesCxxExceptionsInSetUpTestCase(self):}
\DoxyCodeLine{00144\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'C++\ exception\ with\ description\ "{}Standard\ C++\ exception"{}'}}
\DoxyCodeLine{00145\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'\ thrown\ in\ SetUpTestCase()'}}
\DoxyCodeLine{00146\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00147\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInConstructorTest::TearDownTestCase()\ '}}
\DoxyCodeLine{00148\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00149\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00150\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInSetUpTestCaseTest\ constructor\ '}}
\DoxyCodeLine{00151\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00152\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00153\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInSetUpTestCaseTest\ destructor\ '}}
\DoxyCodeLine{00154\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00155\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00156\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInSetUpTestCaseTest::SetUp()\ '}}
\DoxyCodeLine{00157\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00158\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00159\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInSetUpTestCaseTest::TearDown()\ '}}
\DoxyCodeLine{00160\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00161\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00162\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInSetUpTestCaseTest\ test\ body\ '}}
\DoxyCodeLine{00163\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00164\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00165\ }

\end{DoxyCode}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a6e65997bc7dd3700c29fee2e537a9d49}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testCatchesCxxExceptionsInTearDown@{testCatchesCxxExceptionsInTearDown}}
\index{testCatchesCxxExceptionsInTearDown@{testCatchesCxxExceptionsInTearDown}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testCatchesCxxExceptionsInTearDown()}{testCatchesCxxExceptionsInTearDown()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a6e65997bc7dd3700c29fee2e537a9d49} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Catches\+Cxx\+Exceptions\+In\+Tear\+Down (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00189}{189}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00189\ \ \ \textcolor{keyword}{def\ }testCatchesCxxExceptionsInTearDown(self):}
\DoxyCodeLine{00190\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'C++\ exception\ with\ description\ "{}Standard\ C++\ exception"{}'}}
\DoxyCodeLine{00191\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'\ thrown\ in\ TearDown()'}}
\DoxyCodeLine{00192\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00193\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInTearDownTest::TearDownTestCase()\ '}}
\DoxyCodeLine{00194\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00195\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00196\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInTearDownTest\ destructor\ '}}
\DoxyCodeLine{00197\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00198\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00199\ }

\end{DoxyCode}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a0c93888c01978d87a6c57c49b95f7624}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testCatchesCxxExceptionsInTearDownTestCase@{testCatchesCxxExceptionsInTearDownTestCase}}
\index{testCatchesCxxExceptionsInTearDownTestCase@{testCatchesCxxExceptionsInTearDownTestCase}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testCatchesCxxExceptionsInTearDownTestCase()}{testCatchesCxxExceptionsInTearDownTestCase()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a0c93888c01978d87a6c57c49b95f7624} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Catches\+Cxx\+Exceptions\+In\+Tear\+Down\+Test\+Case (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00166}{166}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00166\ \ \ \textcolor{keyword}{def\ }testCatchesCxxExceptionsInTearDownTestCase(self):}
\DoxyCodeLine{00167\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'C++\ exception\ with\ description\ "{}Standard\ C++\ exception"{}'}}
\DoxyCodeLine{00168\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'\ thrown\ in\ TearDownTestCase()'}}
\DoxyCodeLine{00169\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00170\ }

\end{DoxyCode}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_aae7ffa4b9583e66b08b41c5e7317d4a3}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testCatchesCxxExceptionsInTestBody@{testCatchesCxxExceptionsInTestBody}}
\index{testCatchesCxxExceptionsInTestBody@{testCatchesCxxExceptionsInTestBody}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testCatchesCxxExceptionsInTestBody()}{testCatchesCxxExceptionsInTestBody()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_aae7ffa4b9583e66b08b41c5e7317d4a3} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Catches\+Cxx\+Exceptions\+In\+Test\+Body (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00200}{200}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00200\ \ \ \textcolor{keyword}{def\ }testCatchesCxxExceptionsInTestBody(self):}
\DoxyCodeLine{00201\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'C++\ exception\ with\ description\ "{}Standard\ C++\ exception"{}'}}
\DoxyCodeLine{00202\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'\ thrown\ in\ the\ test\ body'}}
\DoxyCodeLine{00203\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00204\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInTestBodyTest::TearDownTestCase()\ '}}
\DoxyCodeLine{00205\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00206\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00207\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInTestBodyTest\ destructor\ '}}
\DoxyCodeLine{00208\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00209\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00210\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'CxxExceptionInTestBodyTest::TearDown()\ '}}
\DoxyCodeLine{00211\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{stringliteral}{'called\ as\ expected.'}}
\DoxyCodeLine{00212\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00213\ }

\end{DoxyCode}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a69dd04c0f3eaf00738e0c3bfe8f31e68}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testCatchesNonStdCxxExceptions@{testCatchesNonStdCxxExceptions}}
\index{testCatchesNonStdCxxExceptions@{testCatchesNonStdCxxExceptions}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testCatchesNonStdCxxExceptions()}{testCatchesNonStdCxxExceptions()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a69dd04c0f3eaf00738e0c3bfe8f31e68} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Catches\+Non\+Std\+Cxx\+Exceptions (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00214}{214}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00214\ \ \ \textcolor{keyword}{def\ }testCatchesNonStdCxxExceptions(self):}
\DoxyCodeLine{00215\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'Unknown\ C++\ exception\ thrown\ in\ the\ test\ body'}}
\DoxyCodeLine{00216\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ EX\_BINARY\_OUTPUT)}
\DoxyCodeLine{00217\ }

\end{DoxyCode}
\Hypertarget{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a5952b0a1b4fd09ba078635bf4e4bb1b1}\index{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}!testUnhandledCxxExceptionsAbortTheProgram@{testUnhandledCxxExceptionsAbortTheProgram}}
\index{testUnhandledCxxExceptionsAbortTheProgram@{testUnhandledCxxExceptionsAbortTheProgram}!gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest@{gtest\_catch\_exceptions\_test.CatchCxxExceptionsTest}}
\doxysubsubsection{\texorpdfstring{testUnhandledCxxExceptionsAbortTheProgram()}{testUnhandledCxxExceptionsAbortTheProgram()}}
{\footnotesize\ttfamily \label{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a5952b0a1b4fd09ba078635bf4e4bb1b1} 
gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+Catch\+Cxx\+Exceptions\+Test.\+test\+Unhandled\+Cxx\+Exceptions\+Abort\+The\+Program (\begin{DoxyParamCaption}\item[{}]{self}{}\end{DoxyParamCaption})}



Definition at line \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source_l00218}{218}} of file \mbox{\hyperlink{gtest__catch__exceptions__test_8py_source}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}.


\begin{DoxyCode}{0}
\DoxyCodeLine{00218\ \ \ \textcolor{keyword}{def\ }testUnhandledCxxExceptionsAbortTheProgram(self):}
\DoxyCodeLine{00219\ \ \ \ \ \textcolor{comment}{\#\ Filters\ out\ SEH\ exception\ tests\ on\ Windows.\ Unhandled\ SEH\ exceptions}}
\DoxyCodeLine{00220\ \ \ \ \ \textcolor{comment}{\#\ cause\ tests\ to\ show\ pop-\/up\ windows\ there.}}
\DoxyCodeLine{00221\ \ \ \ \ FITLER\_OUT\_SEH\_TESTS\_FLAG\ =\ FILTER\_FLAG\ +\ \textcolor{stringliteral}{'=-\/*Seh*'}}
\DoxyCodeLine{00222\ \ \ \ \ \textcolor{comment}{\#\ By\ default,\ Google\ Test\ doesn't\ catch\ the\ exceptions.}}
\DoxyCodeLine{00223\ \ \ \ \ uncaught\_exceptions\_ex\_binary\_output\ =\ \mbox{\hyperlink{classgtest__test__utils_1_1_subprocess}{gtest\_test\_utils.Subprocess}}(}
\DoxyCodeLine{00224\ \ \ \ \ \ \ \ \ [EX\_EXE\_PATH,}
\DoxyCodeLine{00225\ \ \ \ \ \ \ \ \ \ NO\_CATCH\_EXCEPTIONS\_FLAG,}
\DoxyCodeLine{00226\ \ \ \ \ \ \ \ \ \ FITLER\_OUT\_SEH\_TESTS\_FLAG],}
\DoxyCodeLine{00227\ \ \ \ \ \ \ \ \ env=environ).output}
\DoxyCodeLine{00228\ }
\DoxyCodeLine{00229\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'Unhandled\ C++\ exception\ terminating\ the\ program'}}
\DoxyCodeLine{00230\ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \textcolor{keywordflow}{in}\ uncaught\_exceptions\_ex\_binary\_output)}
\DoxyCodeLine{00231\ \ \ \ \ self.assert\_(\textcolor{stringliteral}{'unexpected'}\ \textcolor{keywordflow}{not}\ \textcolor{keywordflow}{in}\ uncaught\_exceptions\_ex\_binary\_output)}
\DoxyCodeLine{00232\ }
\DoxyCodeLine{00233\ }

\end{DoxyCode}
Here is the call graph for this function\+:\nopagebreak
\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[width=350pt]{classgtest__catch__exceptions__test_1_1_catch_cxx_exceptions_test_a5952b0a1b4fd09ba078635bf4e4bb1b1_cgraph}
\end{center}
\end{figure}


The documentation for this class was generated from the following file\+:\begin{DoxyCompactItemize}
\item 
libraries/fc/include/fc/crypto/webauthn\+\_\+json/thirdparty/gtest/googletest/test/\mbox{\hyperlink{gtest__catch__exceptions__test_8py}{gtest\+\_\+catch\+\_\+exceptions\+\_\+test.\+py}}\end{DoxyCompactItemize}
